src/call/chat.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * This file is responsible for all chat
+ * and emoji reaction functionality.
+ */
+
+import { getCallContainerEle } from './call.js';
+import rand from './util.js';
+
+/**
+ * Sets up the in-call chat form, including relevant handlers.s
+ * @param {DailyCall} callObject
+ * @param {func(DailyCall, {})} updateRecording
+ */
+export function setupChatForm(callObject, updateRecording) {
+  setupEmojiReactions(callObject, updateRecording);
+
+  // Retrieve chat form and chat input element
+  const chatForm = getChatForm();
+  const chatInput = chatForm.getElementsByTagName('textarea')[0];
+
+  // Set up form submission handler
+  chatForm.onsubmit = (ev) => {
+    ev.preventDefault();
+    try {
+      // Retrieve chat input value and reset the form
+      const msg = chatInput.value;
+      chatForm.reset();
+
+      // Deduce local participant's user name, if available
+      const name = callObject.participants().local.user_name || 'Guest';
+
+      // Send retrieved chat message to all other participants
+      callObject.sendAppMessage({ kind: 'chat', msg }, '*');
+
+      // Display locally sent chat message
+      addChatMsg(name, msg);
+
+      // Invoke recording update callback with chat payload
+      updateRecording(callObject, {
+        kind: 'chat',
+        name,
+        msg,
+      });
+    } catch (e) {
+      console.error('Failed to send chat message', e);
+    }
+  };
+
+  // Let participant submit a chat message with "Enter" key
+  // This will invoke the "onsubmit" handler above
+  chatInput.addEventListener('keydown', (ev) => {
+    if (ev.key === 'Enter') {
+      chatForm.dispatchEvent(new Event('submit'));
+    }
+  });
+}
+
+/**
+ * Adds a chat message to the message box in the DOM
+ * @param {string} name
+ * @param {string} msg
+ */
+export function addChatMsg(name, msg) {
+  const messagesEle = getMessagesEle();
+  const msgContainerEle = document.createElement('div');
+
+  const nameEle = document.createElement('span');
+  nameEle.className = 'chatName';
+  nameEle.innerText = `${name}: `;
+
+  const msgEle = document.createElement('span');
+  msgEle.className = 'chatMsg';
+  msgEle.innerText = msg;
+
+  msgContainerEle.append(nameEle, msgEle);
+  messagesEle.appendChild(msgContainerEle);
+}
+
+/**
+ * Shows a reaction emoji in the dom
+ * @param {string} emoji
+ */
+export function showReaction(emoji) {
+  const ele = document.createElement('div');
+  ele.innerText = emoji;
+  ele.className = 'reaction';
+  const container = getCallContainerEle();
+  container.append(ele);
```

Comment:
I thought about making this look "nicer" with flying emojis and having  them float up from the screen, but decided to keep the UI of it fairly close to the end result we'll see in the recording to have more of a 1-1 mapping between the app and the recording output.

Reply:
I like the idea of the app and recording animations looking similar!